### PR TITLE
feat(shared): add shared tools and types

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

Move a bunch of shared stuff over from lotus so we can work with things internally

# Implementation

- We have a new tool to do cbor-gen -- cbor-gen-for which allows you to just put a `//go:generate cbor-gen-for ~structName~` comment near the struct that needs cbor methods and then just run `go generate ./...`
- Copied over the shared types I was aware of from RetrievalMarket --Signature, SignedVoucher, Address, BigInt (now TokenAmount)
- Ran go fmt
- Copied over some shared utils - cborutil, crypto
- Removed any lotus references

### CODE REVIEW

File: datatransfer/impl/graphsync/graphsync_impl.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ const (
 	ExtensionDataTransfer = graphsync.ExtensionName("fil/data-transfer")
 )
 
+//go:generate cbor-gen-for ExtensionDataTransferData
```
Comment (ingar) [inline datatransfer/impl/graphsync/graphsync_impl.go]: This is pretty cool.

File: filestore/filestore.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ func NewLocalFileStore(basedirectory string) (FileStore, error) {
 			break
 		}
 	}
-	base := basedirectory[0:i + 1]
+	base := basedirectory[0 : i+1]
```
Comment (ingar) [inline filestore/filestore.go]: There's like `path.Clean`, `path.Join` that can do some of the path manipulation in this file...

File: go.mod [hunk 3]
```diff
@@ -21,27 +16,43 @@ require (
...
+	github.com/multiformats/go-multiaddr-net v0.1.0
+	github.com/multiformats/go-multihash v0.0.9
+	github.com/multiformats/go-varint v0.0.1
+	github.com/polydawn/refmt v0.0.0-20190809202753-05966cbd336a
 	github.com/stretchr/testify v1.4.0
-	github.com/whyrusleeping/cbor-gen v0.0.0-20191116002219-891f55cd449d
-	golang.org/x/sys v0.0.0-20190826190057-c7b8b68b1456 // indirect
+	github.com/whyrusleeping/cbor-gen v0.0.0-20191212224538-d370462a7e8a
 	golang.org/x/xerrors v0.0.0-20191011141410-1b5146add898
+	google.golang.org/appengine v1.5.0
```
Comment (ingar) [inline go.mod]: appengine!

File: datatransfer/impl/graphsync/graphsync_impl.go (hunk 1 above)
Comment (hannahhoward) [reply datatransfer/impl/graphsync/graphsync_impl.go]: wait till it breaks though...

**Pull request merged**